core/src/app.rs

```diff
@@ -427,18 +427,20 @@ impl App {
                                 .map(|n| {
                                     env.state.insert(
                                         n.clone(),
-                                        Value::Array(
-                                            map_envs
-                                                .iter()
-                                                .map(|e| match e.state.get(n) {
-                                                    None => Err(anyhow!(
-                                                "Missing block `{}` output, at iteration {}",
-                                                n, e.map.as_ref().unwrap().iteration
-                                            ))?,
-                                                    Some(v) => Ok(v.clone()),
-                                                })
-                                                .collect::<Result<Vec<_>>>()?,
-                                        ),
+                                        BlockResult {
```

Comment:
As per IRL `env`'s type should not be changed.

The types of `e` and `flat` will carry `BlockResult` but in the final update of `env` from `final` you extract the value and only store the values in `env`